app/models/manageiq/providers/inflector.rb

```diff
@@ -0,0 +1,17 @@
+module ManageIQ::Providers::Inflector
+  class ObjectNotNamespacedError < StandardError; end
+
+  def self.provider_name(class_or_instance)
+    klass = class_or_instance.class == Class ? class_or_instance : class_or_instance.class
+    provider_module(klass).name.split('::').last
+  end
+
+  def self.provider_module(klass, original_object = nil)
+    if klass == Object
+      raise ObjectNotNamespacedError, "Cannot get provider module from non namespaced object #{original_object}"
+    end
+
+    klass.parent == ManageIQ::Providers ? klass : provider_module(klass.parent, klass)
+  end
+  private_class_method :provider_module
```

Comment:
:) actually [rails' Module.parent](https://github.com/rails/rails/blob/86d4e189580e84c4b7effc0c3ebd25b4e8dc4fee/activesupport/lib/active_support/core_ext/module/introspection.rb#L7-L32) does it almost like this, but I liked the recursive implementation and the fact we are not comparing strings here.

But, yes, I can make this simpler :+1: